cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -507,11 +507,20 @@ public KafkaAssemblyOperator(Vertx vertx, boolean isOpenShift,
                 }
             }
             return withVoid(zkSetOperations.maybeRollingUpdate(zkDiffs.resource(),
-                    zkAncillaryCmChange || this.clusterCa.certRenewed() || this.clusterCa.certsRemoved()));
+                    zkAncillaryCmChange || this.clusterCa.certRenewed() || this.clusterCa.certsRemoved(), false));
         }
 
         Future<ReconciliationState> zkScaleUp() {
-            return withVoid(zkSetOperations.scaleUp(namespace, zkCluster.getName(), zkCluster.getReplicas()));
+            int currentReplicas = 0;
+            StatefulSet ss = zkSetOperations.get(namespace, ZookeeperCluster.zookeeperClusterName(name));
+            if (ss != null) {
+                currentReplicas = ss.getSpec().getReplicas();
+            }
+            Future<Integer> result = Future.succeededFuture(currentReplicas);
```

Comment:
I think you can replace all this with `Future<Integer> result = zkSetOperations.getAsync().map(ss -> ss.getSpec().getReplicas());`